src/features/loading.ts

```diff
@@ -85,337 +76,333 @@ const $onClick = Symbol('onClick');
             }
-          } else {
-            this[$showPoster]();
-          }
-        }
 
-        [$showPoster]() {
-          const posterContainerElement = this[$posterContainerElement];
-          const defaultPosterElement = this[$defaultPosterElement];
-          const posterContainerOpacity = parseFloat(
-              self.getComputedStyle(posterContainerElement).opacity!);
-
-          defaultPosterElement.tabIndex = 1;
-          defaultPosterElement.removeAttribute('aria-hidden');
-          posterContainerElement.classList.add('show');
-
-          if (posterContainerOpacity < 1.0) {
-            posterContainerElement.addEventListener('transitionend', () => {
-              this[$modelIsVisible] = false;
-              this[$announceModelVisibility](false);
-            }, {once: true});
-          }
-        }
-
-        [$hidePoster]() {
-          const posterContainerElement = this[$posterContainerElement];
-          const defaultPosterElement = this[$defaultPosterElement];
-
-          if (posterContainerElement.classList.contains('show')) {
-            posterContainerElement.classList.remove('show');
-
-            // We might need to forward focus to our internal canvas, but that
-            // cannot happen until the poster has completely transitioned away
-            posterContainerElement.addEventListener('transitionend', () => {
-              this[$announceModelVisibility](true);
-              requestAnimationFrame(() => {
-                this[$modelIsVisible] = true;
-
-                const root = this.getRootNode();
-
-                // If the <model-viewer> is still focused, forward the focus to
-                // the canvas that has just been revealed
-                if (root &&
-                    (root as Document | ShadowRoot).activeElement === this) {
-                  this[$canvas].focus();
-                }
-
-                // Ensure that the poster is no longer focusable or visible to
-                // screen readers
-                defaultPosterElement.setAttribute('aria-hidden', 'true');
-                defaultPosterElement.removeAttribute('tabindex');
-              });
-            }, {once: true});
-          }
-        }
-
-        async[$updateSource]() {
-          if (this[$modelIsReadyForReveal]) {
-            this[$sourceUpdated] = true;
-            await super[$updateSource]();
-          }
-        }
+            // Ensure that the poster is no longer focusable or visible to
+            // screen readers
+            defaultPosterElement.setAttribute('aria-hidden', 'true');
+            defaultPosterElement.removeAttribute('tabindex');
+          });
+        }, {once: true});
       }
+    }
+
+    [$getModelIsVisible]() {
+      return super[$getModelIsVisible]() || this[$modelIsVisible];
+    }
+
+    [$getLoaded]() {
+      const src = this.src;
+      return super[$getLoaded]() ||
+          !!(src && CachingGLTFLoader.hasFinishedLoading(src));
```

Comment:
I know the logic of these functions is unchanged, but it's still odd to me. Why super or this?

Reply:
I agree, the logic here is somewhat inscrutable. The intention is to express the logic, "true if the model is already known to be loaded by the super class, or is known to be loaded by the loader." And, the reason why those states might not line up is kind of dubious and deserves critical review.